apollo-runtime/src/main/java/com/apollographql/apollo/cache/http/HttpCacheControl.java

```diff
@@ -1,11 +1,39 @@
 package com.apollographql.apollo.cache.http;
 
+/**
+ * HttpCacheControl class represents the cache controlling strategies for reading http responses from the cache when a
+ * network request is made.
+ */
 public enum HttpCacheControl {
+
+  /**
+   * Signals the apollo client to first fetch the data from the http cache. If the data is not present in
+   * the cache, then it is fetched from the network.
+   */
   CACHE_FIRST("cache-first"),
+  /**
+   * Signals the apollo client to only fetch the data from the http cache. If the data is not present in the cache, an
+   * exception is thrown.
+   */
   CACHE_ONLY("cache-only"),
+  /**
+   * Signals the apollo client to only fetch the data from the network request. If the network request fails, an
+   * exception is thrown.
+   */
   NETWORK_ONLY("network-only"),
+  /**
+   * Signals the apollo client to first fetch the data from the network request. If the network request fails, then
+   * the data is fetched from the http cache. If the data in the cache has become stale, an exception is thrown.
+   */
   NETWORK_FIRST("network-first"),
+  /**
+   * Signals the apollo client to first fetch the data from the network request. If the network request fails, then
+   * the data is fetched from the http cache even if it has become stale.
+   */
   NETWORK_BEFORE_STALE("network-before-stale"),
+  /**
+   * Signals the apollo client to mark the data in the cache stale whenever a read takes place next.
+   */
   EXPIRE_AFTER_READ("expire-after-read");
```

Comment:
no, that's right. Http cache for this request will be deleted after the first read